Update to webpack2.

moimael opened the pull request:
Remove unused json-loader, fix eslint errors,
move webpack and babel-loader in devDependencies.

I can add tree-shaking support once pr #590 is merged.

Commit:
Update to webpack2.

Remove unused json-loader, fix eslint errors,
move webpack and babel-loader in devDependencies.

Signed-off-by: Maël Lavault <mael.lavault@universalavenue.com>

avahowell commented:
Tree-shaking would be great to have.

Commit:
Add tree-shaking

Signed-off-by: Maël Lavault <mael.lavault@mailz.org>

moimael commented:
@johnathanhowell Added :)

moimael commented:
Ah seems like tests use import too and not webpack, i'll add env support

Commit:
Fix import not working in test with tree shaking.

Signed-off-by: Maël Lavault <mael.lavault@mailz.org>

moimael commented:
On a side note I have a few other changes in mind, where can discuss this with sia dev ? Do you have a chat somewhere ? Thanks !

lukechampine commented:
@moimael: We are most active in our Slack channel: https://slackin.sia.tech
John is the main UI dev. His Slack handle is johnhowell.

avahowell commented:
Looks like the production webpack config is broken:

```
ERROR in ./plugins/Files.js from UglifyJs
Unexpected token: name (disabled) [./plugins/Files.js:38867,4]

ERROR in ./plugins/Wallet.js from UglifyJs
Unexpected token: name (disabled) [./plugins/Wallet.js:38867,4]

ERROR in ./plugins/Hosting.js from UglifyJs
Unexpected token: name (disabled) [./plugins/Hosting.js:38867,4]

ERROR in ./plugins/Terminal.js from UglifyJs
Unexpected token: name (disabled) [./plugins/Terminal.js:38648,4]

ERROR in ./plugins/Logs.js from UglifyJs
Unexpected token: name (disabled) [./plugins/Logs.js:38648,4]

ERROR in renderer.js from UglifyJs
Unexpected token: name (homePluginView) [renderer.js:83829,5]

ERROR in ./plugins/About.js from UglifyJs
Unexpected token: name (disabled) [./plugins/About.js:33608,4]

ERROR in main.js from UglifyJs
Unexpected token: name (config) [main.js:920,5]
```

avahowell commented:
Would also love to hear about the changes you had in mind, feel free to pop into our slack and shoot me a PM

moimael commented:
Ok the errors come from uglifyjs not supporting es6. The best solution could be to use babilli instead, otherwise we would have to compile everything to es5 to uglify.
I'll PM you ;)

Commit:
Fix production build.

- Use uglifyjs-webpack-plugin instead of bundled webpack uglifyjs plugin
because it allows us to override uglifyjs version
- Use uglifyjs 2.8 harmony version
(we can use uglify-es as soon as webpack support it)
- Remove -p webpack flag for prod build
and declare optimisations explicitly instead

Signed-off-by: Maël Lavault <mael.lavault@mailz.org>

Commit:
Fix webpack warnings by removing dead code.

Signed-off-by: Maël Lavault <mael.lavault@mailz.org>
